pkg/proxy/iptables/proxier.go

```diff
@@ -175,7 +175,7 @@ func newServiceInfo(serviceName proxy.ServicePortName, port *api.ServicePort, se
 
 	if info.onlyNodeLocalEndpoints {
 		p := apiservice.GetServiceHealthCheckNodePort(service)
-		if p == 0 {
+		if p == 0 && service.Spec.Type == api.ServiceTypeLoadBalancer {
```

Comment:
And I'm also changing the service validation to disallow NodePort services to set HealthCheckNodePort.